core/src/main/resources/org/apache/spark/ui/static/log-view.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var baseParams
+
+var curLogLength
+var startByte
+var endByte
+var totalLogLength
+
+var byteLength
+var btnHeight = 30
+
+function setLogScroll(oldHeight) {
+  $(".log-content").scrollTop($(".log-content")[0].scrollHeight - oldHeight);
+}
+
+function tailLog() {
+  $(".log-content").scrollTop($(".log-content")[0].scrollHeight);
+}
```

Comment:
It's not good to repeat a query for the same dom-objects like `$(.log-content)`.